lib/public/components/Log/log.js

```diff
@@ -0,0 +1,160 @@
+import { h } from '/js/src/index.js';
+import { markdownDisplay } from '../common/markdown/markdown.js';
+
+import { formatTimestamp } from '../../utilities/formatting/formatTimestamp.js';
+import { detailsFrontLinks } from '../common/navigation/frontLinks.js';
+
+import { logLinkButton } from './logLinkButton.js';
+import { logReplyButton } from './logReplyButton.js';
+
+/**
+ * Provides formatting for each log field to display that field on the webpage
+ * @param {string} authenticationToken token to authenticate api requests. Used for attachments
+ * @return {Object} an object determining how each log field wil lbe displayed on the page
+ */
+const logFields = (authenticationToken) => ({
+    author: {
+        name: 'Source',
+        format: (author) => author.name,
+    },
+    createdAt: {
+        name: 'Created',
+        format: (timestamp) => formatTimestamp(timestamp),
+    },
+    tags: {
+        name: 'Tags',
+        format: (tags) => h('div.flex-row.flex-wrap.g1', tags.map(({ text }) => h('div.badge.white.bg-gray-light.black', text))),
+    },
+    runs: {
+        name: 'Runs',
+        format: (runs) => detailsFrontLinks(runs, ({ runNumber, id }) => ({
+            content: runNumber,
+            page: 'run-detail',
+            parameters: { id },
+        })),
+    },
+    attachments: {
+        name: 'Attachments',
+        visible: true,
+        format: (attachments) => detailsFrontLinks(
+            attachments,
+            ({ fileName, originalName, id }) => ({
+                content: originalName,
+                href: `/api/attachments/${id}?token=${authenticationToken}`,
+                attributes: { download: fileName, class: 'break-all' },
+            }),
+            { bypassRouter: true },
+        ),
+    },
+});
+
+/**
+ * For some given fields, returns the hyperscript to display those fields in the form
+ * <name>: <formatted data>
+ * @param {Log} log the log
+ * @param {string[]} fields the names of the fields to display
+ * @param {Object} fieldFormats an object containing a format function for each field
+ * @return {Component} the log field displayed
+ */
+const logDetails = (log, fields, fieldFormats) => fields.map((key) =>
+    h(`div#log-${log.id}-${key}.flex-row`, [
+        h('div.mr2', `${fieldFormats[key].name}:`),
+        fieldFormats[key].format(log[key]),
+    ]));
+
+/**
+ * Returns a log's content display
+ *
+ * @param {Log} log the log to display
+ * @param {Boolean} isCollapsed an indicator as to whether the log is collapsed
+ * @return {Component} the log's content display
+ */
+const textBox = (log, isCollapsed) => h(
```

Comment:
This component should be extracted in an individual component and should have improved documentation on what the content of the log includes
For example, it is not clear why we have `logContent` and `logDetails` with both being better suited as individual components for future reusesability